include/envoy/event/dispatcher.h

```diff
@@ -199,6 +200,15 @@ class Dispatcher {
    * @return the watermark buffer factory for this dispatcher.
    */
   virtual Buffer::WatermarkFactory& getWatermarkFactory() PURE;
+
+  /**
+   * Sets a tracked object, which is currently operating in this Dispatcher.
```

Comment:
Presumably we decided to not use RAII semantics here to avoid a memory allocation when setting the tracked object, right? Can you add a comment about that to make it clear we are optimizing for performance here?